app/views/automated_tests/_test_script_results.html.erb

```diff
@@ -0,0 +1,38 @@
+<%# ran_by       → |               |                              |%>
+<%# current_user ↓ |               |                              |%>
+<%#                |   'student'   |         'instructor'         |%>
+<%#    .student?   | student tests | instr tests, results tab     |%>
+<%#   !.student?   | XXXXXXXXXXXXX | instr tests, submissions tab |%>
+<% if ran_by == 'student' %>
+  <% marks = grouping.last_student_test_marks %>
+  <% total_marks = assignment.total_student_test_script_marks %>
+<% elsif ran_by == 'instructor' && local_assigns[:submission] %>
+  <% marks = submission.last_instructor_test_marks %>
+  <% total_marks = assignment.total_instructor_test_script_marks %>
+<% else %><%# ran_by == 'instructor' without a submission (not possible yet) %>
+  <% marks = grouping.last_instructor_test_marks %>
+  <% total_marks = assignment.total_instructor_test_script_marks %>
+<% end %>
+<h2><%= t('automated_tests.test_results') %></h2>
+<div class="block_content">
+  <div class="sub_block test_points">
+    <% if results.empty? %>
+      <%= t('automated_tests.no_results') %>
+    <% elsif !current_user.student? || ran_by == 'student' %>
+      <%= raw t('automated_tests.marks_obtained', marking: "#{marks}/#{total_marks}") %>
```

Comment:
Everyone finds this confusing, not just students. You can remove it altogether. Even for the student-run tests.

Reply:
I agree but I don't know how to make it better. That one time when you get some test numbers wrong, it's actually useful to debug.
I will remove it, and in general we should make the mapping between tests and marks more intuitive.

Reply:
Yes, I agree. I could see, for example, having a new column in the test result table that shows "points awarded" or something.